### ISSUE TITLE
Sound Opened can't play anymore after ending

### ISSUE THREAD

**Issue by clement-redison** [superpoweredSDK/Low-Latency-Android-iOS-Linux-Windows-tvOS-macOS-Interactive-Audio-Platform#586]
Hi, 
I don't have that bug before.
After I Open some sound, I try to play them. After the ending of the sound when I try to play it again, nothing is playing. 
Not with all sound but random sound. 
I open 223 sound max, but with 164, 96 and 10 sound I have the problem too. All my sound have a duration of 2-3 s.
Can you help me with that issue ? 
I don't have any error for debug that.
Clement

**Comment by gaborszanto** [comment]
Hi,

sorry, but i don't understand your explanation. Can you share more details? What do you use to play those sounds? How do you handle the "ending"?

**Comment by clement-redison** [comment]
Yep sorry, 
I create 10 AudioPlayer in Android for 10 sound (.wav file) All file have a duration of 2-3 seconds.
All the time after createAudio player and open the 10 file. I have 10 button for play 10 sound I just load. No problem for now when I click in first time. All sound are playing, but if I wait the end of the sound file who is playing, I can click again on the button no sound is playing anymore.
I try to test with more audio player and sound. All the time 1 2 or 3 file can't be play anymore.
I hope you understand my explanation.

**Comment by gaborszanto** [comment]
Under "AudioPlayer" and "createAudio" which APIs do you mean? Can you link to the relevant documentation?

**Comment by clement-redison** [comment]
I'm using the same as the example for android SuperpoweredPlayer : https://github.com/superpoweredSDK/Low-Latency-Android-iOS-Linux-Windows-tvOS-macOS-Interactive-Audio-Platform/tree/master/Examples_Android/SuperpoweredPlayer

**Comment by gaborszanto** [comment]
And in that example where do you find "AudioPlayer" and "createAudio"? That example doesn't have "AudioPlayer" and "createAudio".

**Comment by clement-redison** [comment]
Yep sorry, createAudioPlayer is my function for start initilization : 
 - SuperpoweredInitialize(...)
 - floatBuffer
 - arrayOf SuperpoweredAdvancedAudioPlayer
 - audioIO

after that I use OpenFile function for my 10 player

I just change the Play function for ->setPosition(0, false, false) and ->play(false)
I don't need a play/pause function juste start from begin.
That code works before I update it. 
My last update is very old so I maybe need to change some lines.

**Comment by gaborszanto** [comment]
And how do you handle the "ending"?

**Comment by clement-redison** [comment]
I don't handle the ending I just wait 3 seconds if I click many times on my button I don't have problem. I have only problem when I'm waiting some seconds like the duration of my files.

**Comment by gaborszanto** [comment]
And when you click your button what do you do?

**Comment by clement-redison** [comment]
When I click on a button I use the Play function with index of the AudioPlayer who have the sound of the button.

**Comment by gaborszanto** [comment]
Do you ALWAYS call every player's process() function in the audio processing callback?

**Comment by clement-redison** [comment]
this is my audioProcessing method it's not my code but before I update I don't have any problem, you can maybe find why it's not working now : 
static bool audioProcessing(
        void *__unused clientdata, // custom pointer
        short int *audio,           // buffer of interleaved samples
        int numberOfFrames,         // number of frames to process
        int __unused samplerate     // sampling rate
) {

    bool overallSilence = true;
    bool addToBuffer = false;
    for (int i = 0; i < instrumentCount; i++) {
        for (int j = 0; j < intensityCount; j++) {
            if (player[i][j] != NULL) {
                bool silence = !player[i][j]->process(floatBuffer, addToBuffer,
                                                      (unsigned int) numberOfFrames);
                if (!silence) {
                    overallSilence = false;
                    addToBuffer = true;
                }
            }
        }
    }

    if (!overallSilence) {
        SuperpoweredFloatToShortInt(floatBuffer, audio, (unsigned int) numberOfFrames);
    }
    return !overallSilence;
}

**Comment by gaborszanto** [comment]
that piece looks okay, so the problem may be somewhere else

**Comment by clement-redison** [comment]
I just try to show in the log the value of player->playing when I have the bug, this value is all the time to true. I can wait 1 2 minutes, that value is block on true. Maybe a bug with that ?

**Comment by clement-redison** [comment]
`const char* playing = "true";
    if (!player[index][indexIntensity]->playing) {
playing = "false";
    }
    log_print(ANDROID_LOG_ERROR, "Play", "play value: %s", playing);
    `

**Comment by gaborszanto** [comment]
There are more than 10.000 apps using Superpowered, many of them high profile with player... Nobody reported such problem, so at this point so far i think there might be a bug in your other codes. Such as this weird instrument/intensity index thing.

**Comment by clement-redison** [comment]
I change my function play was : ->setPosition(0, false, false) and ->play(false) to ->setPosition(0,true,false) ->play(false) and it works now. 
So problem with the playback I think. 
Thanks for the help.
